### PR TITLE
Fix #1366: add missing CQL FP number type conversions, unit tests

### LINKED ISSUE #1366: Data API/JsonCodec: floating-point (FP) numeric values (`decimal`, `double`, `float`), testing, impl
Add:

* Unit tests for 3 supported floating-point CQL types (`decimal`, `double`, `float`)
* ITs for the same to verify creation of table with columns of these types, working `InsertOne`, and `findOne` to fetch.

and fix codec implementations if/when issues found.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds tests for 3 CQL floating-point types (`decimal`, `double`, `float`) wrt Codec conversions from JSON.

**Which issue(s) this PR fixes**:
Fixes #1366

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Initial placeholder/WIP -- added basic unit tests which will be failing until adding 2 x 3 codecs to handle integer-sourced numbers.

**Comment by tatu-at-datastax** [comment]
@amorton Minimal implementation: good enough to unblock your test failures wrt missing codecs for int-to-FP conversions. Needs to have bounds checks for some cases -- was thinking a follow-up PR after refactorings?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java [hunk 1]
```diff
@@ -156,27 +156,76 @@ public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> codecToJSON(DataType targetCQ
...
+          GenericType.LONG,
+          DataTypes.DOUBLE,
+          JSONCodec.ToCQL.safeNumber(Long::doubleValue),
+          JSONCodec.ToJSON.unsafeNodeFactory(JsonNodeFactory.instance::numberNode));
+
+  public static final JSONCodec<BigDecimal, Float> FLOAT_FROM_BIG_DECIMAL =
       new JSONCodec<>(
           GenericType.BIG_DECIMAL,
           DataTypes.FLOAT,
+          // TODO: bounds checks (over/underflow)
```
Comment (amorton) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java:209]: in my testing the under / overflow turns into a +/- Infinity

**Review by amorton — APPROVED** [review]
approved, only thought I had was comments explaining why we have so many conversions typea for floats / decimal but I think it's kind of clear.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/JSONCodecRegistry.java:209]: @amorton Ok: that seems reasonable -- but can discuss if needs different handling

**Comment by tatu-at-datastax** [comment]
> approved, only thought I had was comments explaining why we have so many conversions typea for floats / decimal but I think it's kind of clear.

Yes, need to think of best way to document that "trinity" of values, conversions.
Probably in that container class your refactoring adds.

**Pull request merged**